core/app/models/spree/shipment.rb

```diff
@@ -386,6 +388,33 @@ def manifest_unstock(item)
       stock_location.unstock item.variant, item.quantity, self
     end
 
+    def cancel_inventory_units
+      inventory_units.where.not(state: 'canceled').each(&:cancel!)
+    end
+
+    def resume_inventory_units
```

Comment:
We discussed this within the core team. We think that resuming concerns should be moved into a separate PR. There are a lot of opinions on how resuming an order should work and probably none of them will work as the majority of stores expect (everyone wants to restock items in different ways, depending on if they are using Solidus to manage stocks, which itself has a lot of variables, or they are relying on an external system). 

We are thinking to actually deprecate the Resume order functionality. It's not used a lot and gives us more headaches than the benefits it provides to stores (none of us can think of a single store that ever used this functionality). The way to go here is moving the current behavior into a legacy extension, and deprecate the functionality in core (removing the Resume button from the admin UI). WDYT?

Reply:
I agree with you, especially if it's an underused functionality. 
My solution is far too general and makes a lot of assumptions about the stock management system.

I think that extracting the resume functionality into an external gem, maybe with this fix on it, would be the best solution.

Probably we can close this PR, I'll wait for the final decision about the resume functionality and will come back to contribute to that one as soon as it's ready.

Thank a lot for your feedback and consideration 🙏.

Reply:
The cancellation part of this PR is well done and something that we definitely need in core. Please don't close the PR but just remove the part that resumes the inventory units. 🙏